src/Orckestra.Composer.Cart/Providers/Order/EditingOrderProvider.cs

```diff
@@ -53,6 +54,39 @@ public virtual async Task<bool> CanEdit(Overture.ServiceModel.Orders.Order order
             return isOrderEditable;
         }
 
+        public virtual async Task<bool> IsOrderCancelable(Overture.ServiceModel.Orders.Order order)
+        {
+            if (order.OrderStatus.Equals("Canceled", StringComparison.InvariantCultureIgnoreCase)) return false;
```

Comment:
use this check
if (order?.Cart?.Shipments == null ||
                OrderHistoryConfiguration.CompletedOrderStatuses.Contains(order.OrderStatus))
            {
                return false;
            }